poplar_recovery_builder.sh

```diff
@@ -100,35 +100,45 @@ function parseargs() {
 	# Make sure a single argument was supplied
 	[ $# -lt 1 ] && usage "no partition specified"
 
+	IMAGE_TYPE=Linux
 	INPUT_FILES="L_LOADER USB_LOADER"
 	INPUT_FILES="${INPUT_FILES}"
 
-	if [ "$1" = "all" ] || [ "$1" = "system" ] ; then
-		[ -z "$2" ] && usage "no arg (filesystem) supplied"
-		ROOT_FS_ARCHIVE=$2
-	fi
-
 	case $1 in
-	all|layout|loader|boot|system)
+	all|system)
+		#echo "case all|system" >&2
+		[ -z "$2" ] && usage "no arg (filesystem) supplied"
+		if [ "$2" = "android" ]; then
+			IMAGE_TYPE=Android
+			INPUT_FILES="${INPUT_FILES} ANDROID_BOOT_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_SYSTEM_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_CACHE_IMAGE"
+			INPUT_FILES="${INPUT_FILES} ANDROID_USER_DATA_IMAGE"
+		else
+			ROOT_FS_ARCHIVE=$2
+			INPUT_FILES="${INPUT_FILES} ROOT_FS_ARCHIVE"
+		fi
+		;;&
```

Comment:
I've never seen ";;&" before.  Thank  you for teaching me something new about bash...

Reply:
Ur welcome. :)